### PR TITLE
Add equalsInAnyOrder Matcher for NSArray

### PR DESCRIPTION
**Pull request by jmburges**
There are a bunch of Gists and blogs on this, but I find myself doing `beSupersetOf` and `hasCountOf` together all the time to get the same effect. Figured I would submit this back in.

Converts the NSArrays to NSSets and compares them so that order doesn't matter.

First PR to Expecta so feedback welcome!!

### CODE REVIEW

**Comment by orta** [comment]
I like this idea, but I feel like the current implementation is lossy:

```
expect(@[@1,@1,@2]).to.equalInAnyOrder(@[@1,@2]);
```

Would end up passing here, but they are not the same arrays, in a different order. Perhaps the implementation could order them both by description, then loop through one array and check if the other arrays item is the same?

**Comment by orta** [comment]
Also, CHANGELOG please :)

**Comment by robb** [comment]
> Perhaps the implementation could order them both by description, then loop through one array and check if the other arrays item is the same?

`NSCountedSet` should do the trick here

File: Expecta/Matchers/EXPMatchers+equalInAnyOrder.m [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+  match(^BOOL{
+    if(!actualIsCompatible) return NO;
+
+    NSSet *actualSet = [NSSet setWithArray:actual];
+    NSSet *expectedSet = [NSSet setWithArray:expected];
+
+    if (![actualSet isEqualToSet:expectedSet]) return NO;
+
+    return YES;
```
Comment (robb) [inline Expecta/Matchers/EXPMatchers+equalInAnyOrder.m]: This is just `return [actualSet isEqualToSet:expectedSet]`, right?

**Comment by robb** [comment]
All in all, I'm not sure I think this is worth adding to expecta. It seems that if you frequently need to compare two arrays independent of order, using an `NSArray` is the actual problem.

**Comment by jmburges** [comment]
@robb Thanks for doing so much else! :sparkles: 

I end up using this a ton for times where I want to ensure the same exact keys in an `NSDictionary` and `allKeys` returns an `NSArray`.

I 100% see your point, if you don't care about ordering stop using an `NSArray`. Up to you really :)

**Comment by orta** [comment]
Yeah, I think I'm onboard with the having as an extension, it could fit quite well in [Expecta-Comparison](https://github.com/kylef/Expecta-Comparison) or it could be it's own independent thing.

**Pull request closed without merging**